### DISCUSSION TITLE
Bot Struggling with roles

### DISCUSSION THREAD

**Discussion by DarkFiree31** [danthe1st/Mee6-bypasser#62] [Support] [answered]
The bot only gives roles to certain people.
It has Administrator and is above all of the roles necessary. It works Perfectly for some, but doesn't work at all for others. I don't know if it's because of the MEE6 rank cards, but my theory is that might not recognize spaces, numbers or special symbols. 
Is there a way to solve this?

**Comment by danthe1st** [comment, marked as answer]
The bot uses image recognition. If the card has the same colour as the text on the card (username, discriminator, rank), the bot cannot recognize the user.
Furthermore, it is only able to recognize specific (known) characters and if there are other characters in the name, the user is ignored.

Theoretically, the is another (more reliable) method of querying ranks but that would be against the Mee6 ToS.
